### PR TITLE
Move to subcommand

### PR DESCRIPTION
**Pull request by tych0**
We'd like to move to a `qtile <subcommand>` type command architecture instead of just dumping tons of stuff in /usr/bin. Here's a start on that.

The motivation for this for me was really https://github.com/qtile/qtile/issues/1773

### CODE REVIEW

**Comment by tych0** [comment]
FWIW, I think this no longer a WIP and is ready to go.

File: libqtile/scripts/main.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    start.add_subcommand(subparsers)
+    shell.add_subcommand(subparsers)
+    top.add_subcommand(subparsers)
+    run_cmd.add_subcommand(subparsers)
+    cmd_obj.add_subcommand(subparsers)
+    options = parser.parse_args()
+    if not hasattr(options, "func"):
+        parser.print_help()
+        print("running qtile has moved to subcommand 'qtile start'")
+        os.exit(1)
```
Comment (m-col) [inline libqtile/scripts/main.py:34]: `AttributeError: module 'os' has no attribute 'exit'`. Did you mean sys.exit?

Comment (tych0) [reply libqtile/scripts/main.py:34]: I sure did; now why didn't our CI catch this?

**Comment by m-col** [comment]
The most obvious problem we'll face is people updating their package, not updating their .xinitrc or similar (i guess it's fine if they use a login manager that uses .desktop files), and not being sure as to what is going on. Is it reasonable to default `qtile` to `qtile start`?

**Comment by tych0** [comment]
> Is it reasonable to default qtile to qtile start?

That's a great question -- generally I feel like if you run a program without arguments, it shouldn't do anything. What if I print a big deprecated warning so we can make it not do something later, but default it to `qtile start` for now?

**Comment by m-col** [comment]
Yep I agree that is the cleanest way to go.

**Comment by m-col** [comment]
FWIW I just played around with all of the commands and they all seem to be working fine.

**Comment by m-col** [comment]
Maybe also update the xephyr script?

**Comment by tych0** [comment]
Yeah, and the .desktop file too.

I spent a few minutes playing around with argparse and a default subcommand last night and couldn't get it to work :(

**Comment by tych0** [comment]
Ok, I think this has all of the updates we talked about. @ramnes @m-col

**Comment by m-col** [comment]
Running `qtile start` inside my running session stills starts the new instance :(

**Comment by tych0** [comment]
Hrm, do you have all the x properties? It worked for me.

**Comment by tych0** [comment]
Also - it works in the tests :)

**Comment by m-col** [comment]
Would it be possible to print the help message when `qtile help` is called?

Other than that, all else seems good :)

**Comment by tych0** [comment]
Ok - did you get it sorted out? It is totally possible there's bugs here somewhere.

**Comment by m-col** [comment]
Ah yes sorry, forgot to say, I discovered the cause. You've moved the `sys.path`
lines to after the import of libqtile. This makes running `./bin/qtile` from the
git repository import the installed version, rather than the local git copy.
This was the cause of my error, and is problemetic for when testing things
within the git copy, so the `sys.path` prepend needs to be moved up to before
the import. This issue won't affect the installed `qtile` executable though.

**Comment by tych0** [comment]
Ah ha, that's totally unintentional. Let me revert it. Thanks :)

**Pull request merged**